### ISSUE TITLE
[BUG???] wrong shift of the critic and GAE?

### ISSUE THREAD

**Issue by RunpengDai** [verl-project/verl#1960]
Currently, in `dp_critic.py`, the following line is used to slice the value tensor:

```python
values = values[:, -response_length - 1 : -1].squeeze(-1)
```

This means the `values` tensor contains the value of the sequence up to each generated token, but **does not include \$V(\text{prompt})\$** itself.

However, for advantage calculation on the first output token, we need:

$$
A_1 = Q(\text{prompt}, \text{first token}) - V(\text{prompt})
$$

Similarly, in GAE, the baseline for the first output token should also be \$V(\text{prompt})\$. In the current setup, \$V(\text{prompt})\$ is actually located at index `-response_length - 2` and is not included in the sliced `values` tensor. As a result, advantage computation for the first token is incorrect.

---

Following the same logic, in the `compute_gae_advantage_return` function in `core_algos.py`, we have:

```python
for t in reversed(range(gen_len)):
    multi_nextvalues = values[:, t + 1, :] if t < gen_len - 1 else 0.0
    multi_delta = token_level_rewards[:, t].unsqueeze(-1) + gamma * multi_nextvalues - values[:, t, :]
    multi_lastgaelam = multi_delta + gamma * lam * multi_lastgaelam
    multi_advantages_reveserd.append(multi_lastgaelam)
multi_advantage = torch.stack(multi_advantages_reveserd, dim=1).flip(dims=[1])
```

If we consider calculating the advantage for generating EOS token, it should be:

$$
A_\text{eos} = r(S_{T-1}, \text{eos}) - V(S_{T-1})
$$

where $S_{T-1}$ denotes the sentence up till the last token before EOS.

However, in the current code(in the loop when t=gen_len -1), the advantage is computed as:

$$
A_\text{eos} = r(S_{T-1}, \text{eos}) - V(S_{T})
$$

which is off by one position. This suggests that the slicing in `dp_critic.py` should also shift **one position to the left**:

```python
values = values[:, -response_length - 2 : -2].squeeze(-1)
```

to ensure that the values align properly with the GAE computation.

**I’m not sure if the attention mask needs to be revised in the same way to match the shifted values. Would appreciate feedback or clarification on this point.**

**Comment by Kirrito-k423** [comment]
### 中文
根据本次定时任务对 comments 和 timeline 的检查，这个 issue 命中默认超时关闭规则的零评论零引用例外：创建时间为 2025-06-10T23:06:08Z，评论数量为 0，timeline 未发现公开引用、PR/issue cross-reference、commit/PR mention 或其它公开关联互动，且创建距今超过 6 个月。证据：zero_comment_zero_reference_created_at=2025-06-10T23:06:08Z; comments=0; timeline public references/interactions not found。检查来源：https://api.github.com/repos/verl-project/verl/issues/1960/comments 和 https://api.github.com/repos/verl-project/verl/issues/1960/timeline。因此按定时清理标准先关闭；如果作者或社区成员仍能提供新的复现、证据或需求背景，可以继续评论请求重开或创建新 issue。

### English
In this scheduled pass, I checked the issue comments and timeline. This issue matches the zero-comment / zero-reference exception in the default stale-issue closure rule: it was created at 2025-06-10T23:06:08Z, has 0 comments, no public references, PR/issue cross-references, commit/PR mentions, or other public linked activity were found in the timeline, and it is more than 6 months old. Evidence: zero_comment_zero_reference_created_at=2025-06-10T23:06:08Z; comments=0; timeline public references/interactions not found. Checked sources: https://api.github.com/repos/verl-project/verl/issues/1960/comments and https://api.github.com/repos/verl-project/verl/issues/1960/timeline. I am closing it under the scheduled cleanup rule. If the author or community can provide a fresh reproduction, evidence, or requirement context, please comment to request reopening or open a new issue.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
